docs/configuration_files.rst

```diff
@@ -122,3 +122,264 @@ An example of the main configuraiton file:
         reffile4ssb:
             bpmreffile: /bla/bpm.fits
 
+
+
+
+
+.. _instrument:
+
+Instrument
+----------
+
+The name of the instrument whose reference files are being created. Must be one of the JWST instruments. Capitalization is not important.
+
+.. _reflabels:
+
+Reference Labels
+----------------
+
+The list of allowed reference labels. These are simply the names of the reference file creation packages that can be run. In the near future, this list will be generated automatically by searching for the wrapper scripts around plugged-in modules. At that time, reflabels will be removed from the config file.
+
+.. _reftypes:
+
+Reference Types
+---------------
+
+List of the reference file types that may be created. At a minimum, this list must contain the reference file types corresponding to all of the entries in the reflabels_ list.
+
+.. _default_reflabels:
+
+Default Reference Labels
+------------------------
+
+In this section, the user can define which of the reference file creation packages (reflabels_) is the default for each reference type.
+
+.. _dark_pattern:
+
+Dark Pattern
+------------
+
+List of strings that will be used to identify dark current files within the list of input files. mkrefs will look for these strings in the filenames.
+
+.. _flat_pattern:
+
+Flat Pattern
+------------
+
+List of strings that will be used to identify  flat field files within the list of input files. mkrefs will look for these strings in the filenames.
+
+.. _basenamepattern:
+
+Basename Pattern
+----------------
+
+String to use when looking for input files.
+
+.. _dateobs_fitskey:
+
+Date-Obs Fits Header Keyword
+----------------------------
+
+The header keyword in the input files that contains the date of the observation. Dates are needed to enforce rules when pairing darks and flats. For JWST, set this to **DATE-END**.
+
+.. _timeobs_fitskey:
+
+Time-Obs Fits Header Keyword
+----------------------------
+
+The header keyword in the input files that contains the time of the observation. Time are needed to enforce rules when pairing darks and flats. For JWST, set this to **TIME-END**.
+
+.. _mjdobs_fitskey:
+
+Obervation MJD Fits Header Keyword
+----------------------------------
+
+The header keyword in the input files that contains the time of the observation in MJD.
+
+.. _requiredfitskeys:
+
+Required Fits File Header Keywords
+----------------------------------
+
+List of header keywords which must be present in the input files.
+
+.. _optionalfitskeys:
+
+Optional FitsFile Header Keywords
+---------------------------------
+
+List of optional header keywords in the input files. Used for
```

Comment:
guessing "Used for..." will be completed later?

Reply:
Added text from Armin's response.